aviary/mission/flops_based/phases/groundroll_phase.py

```diff
@@ -0,0 +1,202 @@
+from aviary.variable_info.variables import Dynamic
+from aviary.mission.gasp_based.ode.groundroll_ode import GroundrollODE
+
+
+# TODO: support/handle the following in the base class
+# - phase.set_time_options()
+#     - currently handled in level 3 interface implementation
+# - self.external_subsystems
+# - self.meta_data, with cls.default_meta_data customization point
+@register
+class GroundrollPhase(PhaseBuilderBase):
+    '''
+    A phase builder for a two degree of freedom (2DOF) phase.
+    '''
+    __slots__ = ('external_subsystems', 'meta_data')
+
+    # region : derived type customization points
+    _meta_data_ = {}
+
+    _initial_guesses_meta_data_ = {}
+
+    default_name = 'cruise'
+
+    default_ode_class = GroundrollODE
+
+    default_meta_data = _MetaData
+    # endregion : derived type customization points
+
+    def __init__(
+        self, name=None, subsystem_options=None, user_options=None, initial_guesses=None,
+        ode_class=None, transcription=None, core_subsystems=None,
+        external_subsystems=None, meta_data=None
+    ):
+        super().__init__(
+            name=name, core_subsystems=core_subsystems, subsystem_options=subsystem_options, user_options=user_options, initial_guesses=initial_guesses, ode_class=ode_class, transcription=transcription)
+
+        # TODO: support external_subsystems and meta_data in the base class
+        if external_subsystems is None:
+            external_subsystems = []
+
+        self.external_subsystems = external_subsystems
+
+        if meta_data is None:
+            meta_data = self.default_meta_data
+
+        self.meta_data = meta_data
+
+    def build_phase(self, aviary_options: AviaryValues = None):
+        '''
+        Return a new energy phase for analysis using these constraints.
+
+        If ode_class is None, default_ode_class is used.
+
+        If transcription is None, the return value from calling
+        make_default_transcription is used.
+
+        Parameters
+        ----------
+        aviary_options : AviaryValues (<empty>)
+            collection of Aircraft/Mission specific options
+
+        Returns
+        -------
+        dymos.Phase
+        '''
+        phase: dm.Phase = super().build_phase(aviary_options)
+
+        user_options: AviaryValues = self.user_options
+
+        duration_bounds = user_options.get_val('duration_bounds', units='kn')
+        duration_ref = user_options.get_val('duration_ref', units='kn')
+        constraints = user_options.get_val('constraints')
+
+        phase.set_time_options(fix_initial=True, fix_duration=False,
+                               units="kn", name=Dynamic.Mission.VELOCITY,
+                               duration_bounds=duration_bounds, duration_ref=duration_ref)
+
+        phase.set_state_options("time", rate_source="dt_dv", units="s",
+                                fix_initial=True, fix_final=False, ref=1., defect_ref=1., solve_segments='forward')
```

Comment:
are set_time_options and set_state_options flipped here?

Reply:
Nope, in this case we are integrating with respect to velocity, so the "time" in Dymos is a velocity. The actual time is being computed as a state. Maybe these method names could be changed in Dymos, but we are using them as intended here in Aviary

Reply:
hmm, could we wrap them in aviary to rename them for clarity?

Reply:
This is a good idea in general. I'm not certain I'd want to wrap it for just Aviary, but instead change Dymos. Let's not do that in this PR but I've made an issue on the Dymos side after talking it through with Rob: https://github.com/OpenMDAO/dymos/issues/1048

Reply:
good enough for me